lms/services/canvas_api/client.py

```diff
@@ -210,21 +215,37 @@ def list_files(self, course_id):
         # For documentation of this request see:
         # https://canvas.instructure.com/doc/api/files.html#method.files.api_index
 
-        return self._client.send(
+        files = self._client.send(
             "GET",
             f"courses/{course_id}/files",
             params={"content_types[]": "application/pdf"},
             schema=self._ListFilesSchema,
         )
 
+        for file_dict in files:
+            self._canvas_files_svc.upsert(
+                CanvasFile(
+                    consumer_key=self._lti_user.oauth_consumer_key,
+                    tool_consumer_instance_guid=self._lti_user.tool_consumer_instance_guid,
+                    course_id=course_id,
+                    file_id=file_dict["id"],
+                    filename=file_dict["filename"],
+                    size=file_dict["size"],
+                )
+            )
+
+        return files
```

Comment:
I think we might want to combine this into a single call: `self._canvas_files_svc.upsert([CanvasFile(...) for file_dict in files])`. That way the service method (which now accepts multiple `CanvasFile`'s for upserting) can be more efficient and do just a single `SELECT` and a single `UPSERT`, instead of one per file.